contracts/src/v0.8/functions/dev/1_0_0/accessControl/interfaces/ITermsOfServiceAllowList.sol

```diff
@@ -35,6 +35,16 @@ interface ITermsOfServiceAllowList {
    */
   function isBlockedSender(address sender) external returns (bool);
 
+  /**
+   * @notice Get a list of all allowed senders
+   * @dev WARNING: This operation will copy the entire storage to memory, which can be quite expensive. This is designed
+   * to mostly be used by view accessors that are queried without any gas fees. Developers should keep in mind that
+   * this function has an unbounded cost, and using it as part of a state-changing function may render the function
+   * uncallable if the set grows to a point where copying to memory consumes too much gas to fit in a block.
+   * @return addresses - all allowed addresses
+   */
+  function getAllAllowedSenders() external view returns (address[] memory);
```

Comment:
What do you think about adding a paginated version? There would be a getAllowlistSize() and then getAllowlistSlice(start, end)?

Reply:
I actually don't think we should worry about the allow list getting too large.
As a view function called off-chain we shouldn't have any problems.
The risk is calling it on-chain and running out of gas, but that's not our use case.

Reply:
View functions still have a "virtual" gas limit equal to the block gas limit.  An SLOAD takes 800 gas, plus whatever the additional gas cost is to construct the array, so we would exceed Ethereum's 30mil block gas limit at around 30,000 addresses or so.